rules/service.go

```diff
@@ -0,0 +1,473 @@
+	res, err := http.Get(url)
+	if err != nil {
+		return rules, errors.Wrap(ErrKuiperServer, err)
+	}
+	defer res.Body.Close()
+
+	err = json.NewDecoder(res.Body).Decode(&rules)
+	if err != nil {
+		return rules, errors.Wrap(ErrMalformedEntity, err)
+	}
+
+	for i, value := range rules {
+		rules[i].ID = remove(ui.Id, value.ID)
+	}
+
+	return rules, nil
+}
+
+func (re *reService) ViewRule(ctx context.Context, token, name string) (Rule, error) {
+	var rule Rule
+	ui, err := re.auth.Identify(ctx, &mainflux.Token{Value: token})
+	if err != nil {
+		return rule, ErrUnauthorizedAccess
+	}
+
+	name = prepend(ui.Id, name)
+	url := fmt.Sprintf("%s/rules/%s", re.kuiperURL, name)
+
+	res, err := http.Get(url)
+	if err != nil {
+		return rule, errors.Wrap(ErrKuiperServer, err)
+	}
+	if res.StatusCode == http.StatusNotFound {
+		return rule, errors.Wrap(ErrNotFound, err)
+	}
+	defer res.Body.Close()
+
+	err = json.NewDecoder(res.Body).Decode(&rule)
+	if err != nil {
+		return rule, errors.Wrap(ErrMalformedEntity, err)
+	}
+
+	ruleRemove(ui.Id, &rule)
+
+	return rule, nil
+}
+
+func (re *reService) GetRuleStatus(ctx context.Context, token, name string) (map[string]interface{}, error) {
+	var status map[string]interface{}
+
+	ui, err := re.auth.Identify(ctx, &mainflux.Token{Value: token})
+	if err != nil {
+		return status, ErrUnauthorizedAccess
+	}
+	name = prepend(ui.Id, name)
+	url := fmt.Sprintf("%s/rules/%s/status", re.kuiperURL, name)
+
+	res, err := http.Get(url)
+	if err != nil {
+		return status, errors.Wrap(ErrKuiperServer, err)
+	}
+	if res.StatusCode == http.StatusNotFound {
+		return status, errors.Wrap(ErrNotFound, err)
+	}
+	defer res.Body.Close()
+
+	err = json.NewDecoder(res.Body).Decode(&status)
+	if err != nil {
+		return status, errors.Wrap(ErrMalformedEntity, err)
+	}
+
+	for key, val := range status {
+		delete(status, key)
+		status[remove(ui.Id, key)] = val
+	}
+
+	return status, nil
+}
+
+func (re *reService) ControlRule(ctx context.Context, token, name, action string) (string, error) {
+	ui, err := re.auth.Identify(ctx, &mainflux.Token{Value: token})
+	if err != nil {
+		return "", ErrUnauthorizedAccess
+	}
+
+	name = prepend(ui.Id, name)
+	url := fmt.Sprintf("%s/rules/%s/%s", re.kuiperURL, name, action)
+	res, err := http.Post(url, "", nil)
+	if err != nil {
+		return "", errors.Wrap(ErrKuiperServer, err)
+	}
+	result, err := result(res, action, http.StatusOK)
+	if err != nil {
+		return "", err
+	}
+	return result, nil
+}
+
+func prefix(id string) string {
+	return prefixStart + strings.ReplaceAll(id, "-", "") + prefixEnd
+}
+func prepend(id, name string) string {
+	return fmt.Sprintf("%s%s", prefix(id), name)
+}
+
+func remove(id, name string) string {
+	return strings.Replace(name, prefix(id), "", 1)
+}
+
+func indexOf(element string, data []string) int {
+	for k, v := range data {
+		if element == v {
+			return k
+		}
+	}
+	return -1 //not found.
+}
+
+func rulePrepend(id string, rule *Rule) {
+	rule.ID = prepend(id, rule.ID)
```

Comment:
this `rulePrepend` and `ruleRemove` transformations need to be rethinked, as already mentioned as a problem you are depending on the  `from` keyword so other rules are potentially problematic
I think that possible solution is to have additional table that would keep relation of user id, user friendly name, and unique name that will be used by kuiper